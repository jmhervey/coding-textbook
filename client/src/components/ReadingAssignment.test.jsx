import React from 'react';
import { render } from '@testing-library/react';
import ReadingAssignment from './ReadingAssignment';
import fetchMock from "jest-fetch-mock";

// Configure fetch mock
fetchMock.enableMocks();

describe('ReadingAssignment', () => {
  test('Unit Test for the ReadingAssignment component', () => {

    const items = [
      { id: 1, title: 'Article 1', content: 'Content 1' },
      { id: 2, title: 'Article 2', content: 'Content 2' },
    ];

    // Render the ReadingAssignment component
    const { container } = render(<ReadingAssignment items={items} />);

    // Test if the rendered component matches the saved snapshot
    expect(container.firstChild).toMatchSnapshot();
  });
});
