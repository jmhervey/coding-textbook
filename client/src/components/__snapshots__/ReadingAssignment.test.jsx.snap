// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadingAssignment renders the ReadingAssignment component 1`] = `
<div
  style="width: 100vw; height: 100vh; display: flex; flex-direction: row;"
>
  <nav>
    <ul
      class="sc-bcXHqe ivZAkS"
    >
      Textbooks
    </ul>
    <ul
      class="sc-gswNZR heYmLR"
    >
      <li>
        ☐
        Article 1
      </li>
      <li>
        ☐
        Article 2
      </li>
    </ul>
  </nav>
  Loading...
</div>
`;
